qldpc/codes.py

```diff
@@ -639,38 +798,60 @@ def get_one_distance_upper_bound(
             # support of a candidate pauli-type logical operator
             effective_check_matrix = np.vstack([code_z.matrix, word]).view(np.ndarray)
             candidate_logical_op = qldpc.decoder.decode(
-                effective_check_matrix, effective_syndrome, exact=False, **decoder_args
+                effective_check_matrix, effective_syndrome, **decoder_args
             )
 
-            # check whether the decoding was successful
-            actual_syndrome = effective_check_matrix @ candidate_logical_op % 2
+            # check whether decoding was successful
+            actual_syndrome = effective_check_matrix @ candidate_logical_op % self.field.order
             logical_op_found = np.array_equal(actual_syndrome, effective_syndrome)
 
         # return the Hamming weight of the logical operator
-        return candidate_logical_op.sum()
-
-    @cachetools.cached(cache={}, key=lambda self, pauli, **decoder_args: (self, pauli))
-    def get_distance_exact(self, pauli: Literal[Pauli.X, Pauli.Z]) -> int:
-        """Exact X-distance or Z-distance of this code."""
+        return int(np.count_nonzero(candidate_logical_op))
+
+    # TODO: use vector...
```

Comment:
@mittaltushant could you please take a look at this TODO (also the one in `get_distance_from_logical_ops`).  Once we fix these TODOs I'll make a new PR to merge just the new distance-related changes (basically, `codes.py` and `codes_test.py` in this PR).

Reply:
I made a change to the first one. I vaguely remember discussing with you that we would get rid of get_distance_logical_ops as it does not give true minimum weight anyway.

Reply:
We did discuss this and it's correct that it does not give a true minimum, but it seems(?) to perform better than `get_distance_bound`.  On the other hand, that might be simply because `get_distance_logical_ops` essentially takes the best of `2 k` bounds...

Let me think on this a bit.

Reply:
Yeah I think `get_distance_logical_ops` should be no better than `get_distance_bound(num_trials=code.dimension)`.  I removed `get_distance_logical_ops`.